pkgs/development/python-modules/matrix-nio/default.nix

```diff
@@ -16,28 +17,35 @@
 , peewee
 , cachetools
 , atomicwrites
+, poetry-core
 }:
 
 buildPythonPackage rec {
   pname = "matrix-nio";
-  version = "0.15.2";
+  version = "0.17.0";
 
   src = fetchFromGitHub {
     owner = "poljar";
     repo = "matrix-nio";
     rev = version;
-    sha256 = "190xw3cvk4amr9pl8ip2i7k3xdjd0231kn2zl6chny5axx22p1dv";
+    sha256 = "sha256:0fdcbkrki5ni8aaq9xbrmpj1n7hf9z1vvi1b6jkzl55ahc0n50ki";
   };
 
+  format = "pyproject";
+
   nativeBuildInputs = [
     git
+    poetry-core
   ];
 
+  patches = [ ./update-dependencies.patch ];
+
```

Comment:
This could also do with being a `substituteInPlace`, and there's often little point in bumping it up to a specific version as it just means the next person will have to manually bump the bumps next time a dependency is bumped in nixpkgs. Does it work if you switch them all to `= "*"`?

Also it would be nice to have a go at enabling some of the tests (or at least a `pythonImportsCheck`) if we're going to be messing with the dependencies now.